File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int n: nums) map.put(n, map.getOrDefault(n, 0) + 1);
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> map.get(a) - map.get(b));
        for(int key: map.keySet()){
            pq.add(key);
            if(pq.size() == k + 1) pq.poll();
        }
        int[] ans = new int[k];
        for(int i = 0 ; i < k ; i++) ans[i] = pq.poll();
        return ans;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
from collections import defaultdict
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = defaultdict(int)
        for num in nums:
            hashmap[num] +=  1
        keys = list(hashmap.keys())
        keys = sorted(keys, key=lambda k: hashmap[k], reverse=True)
        
        return keys[:k]